### PR TITLE
#65 fix tag name when the registry not on default port

### PR DESCRIPTION
**Pull request by liuhao92**
## what this PR do ?

serach the tag separator after last backSlash,because when the registry is not on default port , such as foobar:8443/foo/bar the tag name will not be processed correctly.

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/gradle-docker](https://github.com/palantir/gradle-docker), @liuhao92! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: src/test/groovy/com/palantir/gradle/docker/PalantirDockerPluginTests.groovy [hunk 1]
```diff
@@ -330,6 +330,38 @@ class PalantirDockerPluginTests extends AbstractPluginTest {
         execCond("docker rmi -f ${id}:another")
     }
 
+    def 'tags are correctly processed'() {
```
Comment (uschi2000) [inline src/test/groovy/com/palantir/gradle/docker/PalantirDockerPluginTests.groovy]: Can you just test the 'computeName' method separately? No need for end-to-ebd here.

File: src/main/groovy/com/palantir/gradle/docker/PalantirDockerPlugin.groovy [hunk 2]
```diff
@@ -146,7 +146,14 @@ class PalantirDockerPlugin implements Plugin<Project> {
     }
 
     private String computeName(String name, String tag) {
-        return name.replaceAll(":.*", "") + ":" + tag
+        def lastBackSlashIndex = name.lastIndexOf("/");
```
Comment (uschi2000) [inline src/main/groovy/com/palantir/gradle/docker/PalantirDockerPlugin.groovy:149]: This method is a little hard to understand. Needs a better name, maybe a comment.

File: src/test/groovy/com/palantir/gradle/docker/PalantirDockerPluginTests.groovy (hunk 1 above)
Comment (liuhao92) [reply src/test/groovy/com/palantir/gradle/docker/PalantirDockerPluginTests.groovy]: OK, I will do that.

Comment (uschi2000) [reply src/test/groovy/com/palantir/gradle/docker/PalantirDockerPluginTests.groovy]: cheers!

Comment (liuhao92) [reply src/test/groovy/com/palantir/gradle/docker/PalantirDockerPluginTests.groovy]: If there is some problem in this PR, Please let me know 
Im the first time to be  contributor , thank you very much.  : )

File: src/main/groovy/com/palantir/gradle/docker/PalantirDockerPlugin.groovy [hunk 3]
```diff
@@ -145,8 +146,17 @@ class PalantirDockerPlugin implements Plugin<Project> {
         }
     }
 
-    private String computeName(String name, String tag) {
-        return name.replaceAll(":.*", "") + ":" + tag
+    @PackageScope
+    String getImageNameByTag(String name, String tag) {
```
Comment (uschi2000) [inline src/main/groovy/com/palantir/gradle/docker/PalantirDockerPlugin.groovy]: static method

File: src/test/groovy/com/palantir/gradle/docker/PalantirDockerPluginTests.groovy [hunk 4]
```diff
@@ -417,5 +417,18 @@ class PalantirDockerPluginTests extends AbstractPluginTest {
         execCond("docker rmi -f ${id}") || true
     }
 
+    def 'check imageName correctly processed'() {
```
Comment (uschi2000) [inline src/test/groovy/com/palantir/gradle/docker/PalantirDockerPluginTests.groovy:420]: can probably be simpler:

```
def '...'() {
  PalantirDockerPlugin.getImageNameByTag('...', '...') == '...'
}
```

Comment (liuhao92) [reply src/test/groovy/com/palantir/gradle/docker/PalantirDockerPluginTests.groovy:420]: Ok, Im a java developer,not be familiar with groovy, I have Learned a lot.

Comment (liuhao92) [reply src/test/groovy/com/palantir/gradle/docker/PalantirDockerPluginTests.groovy:420]: Thank you for your patience to review this.

**Comment by guenhter** [comment]
Is already handled by https://github.com/palantir/gradle-docker/pull/70

IMHO this can be closed.

**Pull request closed without merging**